core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -2808,6 +2810,18 @@ public boolean isQuietingDown() {
         return isQuietingDown;
     }
 
+    /**
+     * Returns quiet down reason if it was indicated.
+     * @return
+     *      Reason if it was indicated. null otherwise
+     *      @since TODO
+     */
+    @Exported
+    @CheckForNull
+    public String getQuietDownReason() {
+        return quietDownReason;
```

Comment:
Wouldn't it be cleaner to use the new `quietDownReason` as the only indicator for quiet down state?

E.g.:

- `null`: Not quieting down
- `""`: Quieting down, no reason
- any other value: Quieting down, this reason

(Or a proper value object with a basic, sane API)

Having two independent variables seems to invite inconsistent states.

Reply:
I agree with you, will refactor it to one indicator.

- [ ] Refactor to one shutdown state field